web/db.py

```diff
@@ -7,17 +7,12 @@
 import os
 import re
 import time
+from urllib import parse as urlparse
+from urllib.parse import unquote
```

Comment:
Sorry to be a pain but...
```suggestion
from urllib.parse import unquote, urlparse
```
and the down around line 1600 change `urlparse.urlparse` --> `urlparse`